File: 0455-assign-cookies/0455-assign-cookies.swift
class Solution {
    func findContentChildren(_ g: [Int], _ s: [Int]) -> Int {
        let sortedG = g.sorted()
        let sortedS = s.sorted()
        var sIdx = 0
        var count = 0
        for (i, value) in sortedG.enumerated() {
            while sIdx < sortedS.count {
                if value <= sortedS[sIdx]{
                    count += 1
                    sIdx += 1
                    break
                }

                sIdx += 1
            }
        }
        return count
    }
}

File: 0455-assign-cookies/0455-assign-cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        int sIdx = 0;

        Arrays.sort(g);
        Arrays.sort(s);

        int content = 0;
        for (int i = 0; i < g.length; i++) {
            while(sIdx < s.length) {
                if(g[i] <= s[sIdx]) {
                    content += 1;
                    sIdx += 1;
                    break;
                }

                sIdx += 1;
            }
        }

        return content;
    }
}